crates/fuel-core/src/query/message.rs

```diff
@@ -120,54 +134,61 @@ pub trait MessageProofData {
     fn transactions_on_block(&self, block_id: &BlockId) -> StorageResult<Vec<Bytes32>>;
     /// Get the signature of a fuel block.
     fn signature(&self, block_id: &BlockId) -> StorageResult<Signature>;
-    /// Get the fuel block.
-    fn block(&self, block_id: &BlockId) -> StorageResult<CompressedBlock>;
-}
 
-impl MessageProofData for MessageQueryContext<'_> {
-    fn receipts(&self, transaction_id: &TxId) -> StorageResult<Vec<Receipt>> {
-        TransactionQueryContext(self.0).receipts(transaction_id)
-    }
+    // Testing methods only which should be removed in subsequent interface changes, as they just bind to other query methods
+    fn msg_block(&self, block_id: &BlockId) -> StorageResult<CompressedBlock>;
 
-    fn transaction(&self, transaction_id: &TxId) -> StorageResult<Transaction> {
-        TransactionQueryContext(self.0).transaction(transaction_id)
-    }
+    fn msg_transaction(&self, transaction_id: &TxId) -> StorageResult<Transaction>;
 
+    fn msg_receipts(&self, transaction_id: &TxId) -> StorageResult<Vec<Receipt>>;
+}
+
+impl<D: DatabasePort> MessageProofData for D {
     fn transaction_status(
         &self,
         transaction_id: &TxId,
     ) -> StorageResult<TransactionStatus> {
-        TransactionQueryContext(self.0).status(transaction_id)
+        self.status(transaction_id)
     }
 
     fn transactions_on_block(&self, block_id: &BlockId) -> StorageResult<Vec<TxId>> {
-        self.block(block_id).map(|block| block.into_inner().1)
+        crate::query::block::BlockQueryData::block(self, block_id)
+            .map(|block| block.into_inner().1)
     }
 
     fn signature(&self, block_id: &BlockId) -> StorageResult<Signature> {
-        let consensus = BlockQueryContext(self.0).consensus(block_id)?;
+        let consensus = self.consensus(block_id)?;
         match consensus {
             // TODO: https://github.com/FuelLabs/fuel-core/issues/816
             Consensus::Genesis(_) => Ok(Default::default()),
             Consensus::PoA(c) => Ok(c.signature),
         }
     }
 
-    fn block(&self, block_id: &BlockId) -> StorageResult<CompressedBlock> {
-        BlockQueryContext(self.0).block(block_id)
+    fn msg_receipts(&self, transaction_id: &TxId) -> StorageResult<Vec<Receipt>> {
+        crate::query::tx::TransactionQueryData::receipts(self, transaction_id)
+    }
+
+    fn msg_transaction(&self, transaction_id: &TxId) -> StorageResult<Transaction> {
+        crate::query::tx::TransactionQueryData::transaction(self, transaction_id)
+    }
+
+    fn msg_block(&self, block_id: &BlockId) -> StorageResult<CompressedBlock> {
+        crate::query::block::BlockQueryData::block(self, block_id)
     }
 }
 
 /// Generate an output proof.
 // TODO: Do we want to return `Option` here?
-pub fn message_proof<Data: MessageProofData>(
-    data: &Data,
+#[allow(clippy::borrowed_box)]
+pub fn message_proof<'a, T: MessageProofData + ?Sized>(
+    data: &Box<T>,
```

Comment:
```suggestion
pub fn message_proof<'a, T: MessageProofData + ?Sized>(
    data: &T,
```

Reply:
Resolved by moving directly to a GQL`Database` reference, sized issues among other things stopped me from doing just `&T`